src/vm/eventpipe.cpp

```diff
@@ -482,22 +482,31 @@ INT_PTR QCALLTYPE EventPipeInternal::CreateProvider(
     return reinterpret_cast<INT_PTR>(pProvider);
 }
 
-INT_PTR QCALLTYPE EventPipeInternal::AddEvent(
+INT_PTR QCALLTYPE EventPipeInternal::DefineEvent(
     INT_PTR provHandle,
-    __int64 keywords,
     unsigned int eventID,
+    __int64 keywords,
     unsigned int eventVersion,
     unsigned int level,
-    bool needStack)
+    void *pMetadata,
+    unsigned int metadataLength)
 {
     QCALL_CONTRACT;
+
+    EventPipeEvent *pEvent = NULL;
+
     BEGIN_QCALL;
 
-    // TODO
+    _ASSERTE(provHandle != NULL);
+    _ASSERTE(pMetadata != NULL);
+    _ASSERTE(metadataLength >= 20);
+    EventPipeProvider *pProvider = reinterpret_cast<EventPipeProvider *>(provHandle);
+    pEvent = pProvider->AddEvent(keywords, eventID, eventVersion, (EventPipeEventLevel)level, (BYTE *)pMetadata, metadataLength);
```

Comment:
I would recommend reordering the arguments here as well (eventID should come before keywords), because eventID really is the most important item.